stackslib/src/chainstate/nakamoto/coordinator/mod.rs

```diff
@@ -475,7 +485,8 @@ pub fn load_nakamoto_reward_set<U: RewardSetProvider>(
             }
 
             match NakamotoChainState::get_nakamoto_tenure_start_block_header(
-                chain_state.db(),
+                &mut chain_state.index_conn(),
+                stacks_tip,
                 &sn.consensus_hash,
             ) {
                 Ok(Some(x)) => return Some(Ok(x)),
```

Comment:
Why is stacks_tip necessary for finding the starting block of a tenure with a given consensus hash?

Reply:
In the event that there are two or more Stacks forks per Bitcoin block, there can be multiple different tenure-start blocks for the same coinbase height.  Passing a tip ensures that at most one such tenure-start block is ever referenced.

Reply:
But this passes a consensus hash, too. So in the event of a stacks fork, one fork would return `Ok(None)` and another `Ok(Some(..))` (i.e., the contents of some could never be different)?

Reply:
Right -- we're getting the start block, which requires knowing (or computing) the block ID.  The consensus hash isn't enough; you also have to know the block hash.

The way this is achieved under the hood is that we add the tenure-start block ID to the MARF and key it to the tenure consensus hash, so given a tip and a consensus hash, we can find the tenure-start block ID (and thus its header).

Reply:
To answer concretely, suppose there are two Stacks forks going on when the sortition leading to consensus hash _C_ is processed.  Suppose fork A includes a tenure-start block for _C_, and suppose fork B does not (i.e. fork B ignores the winning miner, while A confirms it).  Then, in fork A, there is a tenure-start block for _C_, whereas in fork B, there is not.